app.ts

```diff
@@ -0,0 +1,287 @@
+    }
+
+    let check = ["undefined", "string", ""];
+
+    try {
+      if (check.includes(typeof fromNumber) || fromNumber < 0) {
+        alert("Does not exists");
+        recordIdValue.value = "0";
+      } else if (typeof fromNumber === "number" && fromNumber >= 0) {
+        getRecords(fromNumber, toNumber);
+      }
+    } catch (error) {
+      alert(error);
+    }
+  });
+}
+
+function createHeadingGrid(headings: any) {
+  let headingsData: any = `<h1 class="column-heading">${headings}</h1>`;
+  headingColumns.innerHTML += headingsData;
+}
+
+let recordCount = () => {
+  fetch("http://localhost:2050/recordCount", {
+    method: "GET",
+    headers: { "Content-Type": "application/json" },
+  })
+    .then((response) => response.text())
+    .then((data) => {
+      recordNumberTotal = JSON.parse(data);
+    })
+    .catch((error) => {
+      console.log(error);
+    });
+};
+
+let headingRowCreation = () => {
+  fetch("http://localhost:2050/columns", {
+    method: "GET",
+    headers: { "Content-Type": "application/json" },
+  })
+    .then((response) => response.text())
+    .then((data) => {
+      let headingDataList = JSON.parse(data);
+      let headings: string;
+
+      for (let i = 0; i < headingDataList.length; i++) {
+        headings = headingDataList[i];
+        createHeadingGrid(headings);
+      }
+      resizeScreenData();
+    })
+    .catch((error) => {
+      console.log(error);
+    });
+};
+
+function dynamicGrid(columnData: any) {
+  // Creates the row that the info will display and adds it to the infoColumnsArea.
+  let infoDataRow = `<div id="info-row-${columnData[0]}" class="info-rows"></div>`;
+  infoColumns.innerHTML += infoDataRow;
+  // Gets the created rows.
+  let finalInfoDataRow: any = document.querySelector("#info-row-" + columnData[0] + ".info-rows");
+
+  // Loops through
+  for (let x = 0; x < columnData.length; x++) {
+    let infoData = `<p class="info-row-data">${columnData[x]}</p>`;
+    finalInfoDataRow.innerHTML += infoData;
+  }
+}
+
+let resizeScreenData = () => {
+  let toNumber: number;
+  let selectionArea: any = document.querySelector("#record-navigation-container");
+
+  if (selectionArea.contains(document.querySelector(".navigation-btns"))) {
+    let numberOfRows = Math.floor(window.innerHeight / 50);
+
+    nextBtn.disabled = false;
+    previousBtn.disabled = false;
+
+    let finalRecord = recordNumberTotal - 1;
+
+    if (fromNumber + numberOfRows >= finalRecord) {
+      fromNumber = finalRecord - numberOfRows;
+      nextBtn.disabled = true;
+      previousBtn.disabled = false;
+    } else if (fromNumber <= 0) {
+      nextBtn.disabled = false;
+      previousBtn.disabled = true;
+      fromNumber = 0;
+    }
+
+    toNumber = fromNumber + numberOfRows;
+
+    getRecords(fromNumber, toNumber);
+  }
+};
+
+function getRecords(fromNumber: number, toNumber: number) {
+  fetch(`http://localhost:2050/records?from=${fromNumber}&to=${toNumber}`, {
```

Comment:
You should add a `return` in front of your `fetch` to allow other functions to wait on this function.